src/firefly/js/ui/DropDownMenu.jsx

```diff
@@ -82,12 +78,12 @@ DropDownMenuWrapper.propTypes= {
 
 export class DropDownSubMenu extends PureComponent {
 
-    constructor(props) {
+    constructor(props, context) {
         super(props);
         this.state= {showSubMenu: false};
         this.show = this.show.bind(this);
         this.hide = this.hide.bind(this);
-
+        this.dropdownDirection = context.dropdownDirection || DEFAULT_DROPDOWN_DIR;
```

Comment:
`context` might be null, you should use `lodash` `get`, but there is no point in getting it in the constructor.  it just complicates the class.